src/linker/pipeline.py

```diff
@@ -25,72 +21,36 @@ def set_runner(self, runner: Callable) -> None:
     def run(self, results_dir: Path):
         if not self.runner:
             raise RuntimeError("Runner has not been set.")
-        for step_name in self.steps:
-            self.runner(
+        for step in self.steps:
+            step.run(
```

Comment:
I will admit I don't completely understand this, but it seems like here would be a good place for the Pipeline to have direct contact with its Implementations, and run _those_ instead of "steps."

Reply:
@stevebachmeier let's talk during mid-day about this.

Reply:
Separate PR